### PR TITLE
cda fix

### PR DESCRIPTION
**Pull request by grzebzi**
fixed url collection, must delete 4 chars from the end of decoded string

### CODE REVIEW

File: lib/resolveurl/plugins/cda.py [hunk 1]
```diff
@@ -55,6 +55,15 @@ def cda_decode(self, url):
...
         for char in url:
             strurl += chr(33 + (ord(char) + 14) % 94) if 32 < ord(char) < 127 else char
+        strurl = self.da(strurl)
         strurl = strurl if strurl.startswith('http') else 'https://{0}'.format(strurl)
         strurl = strurl if strurl.endswith('.mp4') else '{0}.mp4'.format(strurl)
-        return strurl.replace("adc.mp4", ".mp4")
+        return strurl
+
+    def da(self, a):
+        a=a[:-4]
```
Comment (Gujal00) [inline lib/resolveurl/plugins/cda.py:64]: Are you sure that this line is required? I can't see any slice instance in the javascript code

File: lib/resolveurl/plugins/cda.py [hunk 2]
```diff
@@ -55,6 +55,15 @@ def cda_decode(self, url):
         strurl = ''
         for char in url:
             strurl += chr(33 + (ord(char) + 14) % 94) if 32 < ord(char) < 127 else char
+        strurl = self.da(strurl)
         strurl = strurl if strurl.startswith('http') else 'https://{0}'.format(strurl)
```
Comment (Gujal00) [inline lib/resolveurl/plugins/cda.py:59]: this line is now redundant

File: lib/resolveurl/plugins/cda.py [hunk 3]
```diff
@@ -55,6 +55,15 @@ def cda_decode(self, url):
...
         strurl = strurl if strurl.endswith('.mp4') else '{0}.mp4'.format(strurl)
```
Comment (Gujal00) [inline lib/resolveurl/plugins/cda.py:60]: This line is also redundant

File: lib/resolveurl/plugins/cda.py [hunk 4]
```diff
@@ -55,6 +55,15 @@ def cda_decode(self, url):
...
-        return strurl.replace("adc.mp4", ".mp4")
+        return strurl
+
+    def da(self, a):
+        a=a[:-4]
+        a = a.replace("0)sss.mp4", ".mp4")
```
Comment (Gujal00) [inline lib/resolveurl/plugins/cda.py:65]: Are you sure that this line is required? I can't see any replace instance in the javascript code

**Review by Gujal00 — CHANGES_REQUESTED** [review]

**Comment by Gujal00** [comment]
Thanks for the PR. I have cherry picked the changes and adapted it
https://github.com/jsergio123/script.module.resolveurl/commit/c50789551a350f812da05ddfc9180422475685a8

**Comment by grzebzi** [comment]
Thanks, 
slice is required, I've compared result address and the actual file streamed in browser.
Not sure about replace, that might be some old stuff.

I expect CDA to change coding frequently.

**Comment by Gujal00** [comment]
@grzebzi 
As suspected it is not a slice and hence is broken in current release
I located the logic and it could be 3 or 4 or 5 characters  replaced. The fixed version is in the git now. WIll be included in the next release

**Pull request closed without merging**